core/src/net/tcp_listener.cpp

```diff
@@ -0,0 +1,115 @@
+#ifdef _WIN32
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#pragma comment(lib, "Ws2_32.lib")
+#else
+#include <arpa/inet.h>
+#include <fcntl.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <unistd.h>
+#endif
+
+#include <cstring>
+#include <memory>
+
+#include <cubos/core/log.hpp>
+#include <cubos/core/net/tcp_listener.hpp>
+#include <cubos/core/net/tcp_stream.hpp>
+#include <cubos/core/reflection/external/cstring.hpp>
+#include <cubos/core/reflection/external/primitives.hpp>
+#include <cubos/core/reflection/external/string.hpp>
+
+using cubos::core::net::Address;
+using cubos::core::net::TcpListener;
+using cubos::core::net::TcpStream;
+
+/// @brief Cross-platform utility to get last error message.
+/// @return Message.
+/// @todo Move this to a higher module for general purposes?
```

Comment:
Yeah that might be a good idea, maybe a `logSystemError()` function in `log.hpp`?